PocketTutor-Team/pockettutor-app
# Add the screen to show tutor details when we select a tutor in the matching screen

# Implement SelectedTutorDetails screen to see the details of a tutor we have selected

### PR Description
In this PR, I implemented the SelectedTutorDetails screen, which is used after the TutorMatching screen. This screen is used for students when they create a lesson, match with some tutors and select one for his lesson. It enable the user to see the tutor details and particulary his description and ratings.

This PR also implement tests for the screen to be sure everything is display, and we can correctly use all button in contained in the screen.

### Modifications made
- Implement the new screen composed of separate composable to display tutor information and the best three reviews.
- Add test for the Screen with mocking of the lesson and profile repository and view model
- Add navigation to this screen from the TutorMatching screen (when we click on a tutor card)
- Update the tests of TutorMatching so that it test the navigation to this screen
- Separate the ErrorState component which was implemented in TutorLessonResponse so that it can be also used when the tutor is not found for the SelectedTutorDetails screen

### How to test this PR
There is multiple ways to test if the code of the screen is correct. First of all, I wrote some test that should ensure that everything is display and that we can correctly click on the confirmation button. Another way to test it in a manual way is to start the MainActivity, log-in the application as a student, create a new lesson (with tutor(s) available for it) and click on one available tutor. This will display the screen implemented in the PR and enable you to try whatever you want.

### Screen appearance 
![image](https://github.com/user-attachments/assets/05fdfb91-b43c-423a-a286-30ccff85e590)
